Fika.Core/Networking/Http/FikaRequestHandler.cs

```
@@ -30,32 +36,34 @@ static FikaRequestHandler()
-            string ipString = await client.GetStringAsync("https://api.ipify.org/");
-            ipString = ipString.Replace("\n", "");
-            if (IPAddress.TryParse(ipString, out IPAddress ipAddress))
+            HttpClient client = new HttpClient();
+            IPAddress ipAddress = new IPAddress([0,0,0,0]);
+            foreach (string checkerUrl in checkerUrls)
             {
-                if (ipAddress.AddressFamily is AddressFamily.InterNetwork)
+                string ipString = await client.GetStringAsync(checkerUrl);
+                // some services return IP + newline, some don't
+                ipString = ipString.Replace("\n", "");
+                try
                 {
-                    return ipAddress;
-                }
-                throw new ArgumentException($"IP address was not an IPv4 address, was: {ipAddress.AddressFamily}, address: {ipAddress}!");
-            }
-            ipString = await client.GetStringAsync("https://checkip.amazonaws.com/");
-            if (IPAddress.TryParse(ipString, out ipAddress))
-            {
-                if (ipAddress.AddressFamily is AddressFamily.InterNetwork)
+                    // does the result look like an IP address?
+                    ipAddress = IPAddress.Parse(ipString);
+                    // is the IP address IPv4?
+                    if (ipAddress.AddressFamily is AddressFamily.InterNetwork)
+                    {
+                        return ipAddress;
+                    }
+                } catch (ArgumentNullException)
+                {
+                    // ipString null
+                } catch (FormatException)
                 {
-                    return ipAddress;
+                    // ipString not a valid IP
                 }
-                throw new ArgumentException($"IP address was not an IPv4 address, was: {ipAddress.AddressFamily}, address: {ipAddress}!");
             }
```

I forsee certain issues in debugging as to why things have fucked up here, these should at least throw an exception